src/clib/lib/job_queue/lsf_driver.cpp

```diff
@@ -291,11 +303,11 @@ static char *alloc_quoted_resource_string(const lsf_driver_type *driver) {
 
     char *quoted_resource_request = NULL;
     if (req) {
-        if (driver->submit_method == LSF_SUBMIT_REMOTE_SHELL)
+        if (driver->submit_method == LSF_SUBMIT_REMOTE_SHELL) {
             quoted_resource_request = util_alloc_sprintf("\"%s\"", req);
-        else
-            quoted_resource_request = util_alloc_string_copy(req);
-        free(req);
+            free(req);
```

Comment:
You free the char* only if `LSF_SUBMIT_REMOTE_SHELL`, but on line 269 you free the char* regardless of use.
Is this intentional?

Reply:
Yes, this means that req is stolen to quoted_resource_request instead of having a copy and then free the old version which was the old behavior.